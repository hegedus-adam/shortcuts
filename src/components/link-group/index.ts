export { renderGroup } from './link-group.component';
