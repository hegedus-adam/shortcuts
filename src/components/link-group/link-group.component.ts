import { renderLink } from '../link';

interface IGroup {
  groupId: number
  groupLabel: string
  links: ILink[]
}

interface ILink {
  id: number
  label: string
  url: string
}

export function renderGroup(group: IGroup) {
  const section = document.createElement('section');
  const h2 = document.createElement('h2');
  h2.innerHTML = group.groupLabel;

  const ul = document.createElement('ul');
  group.links.forEach(link => {
    ul.appendChild(renderLink(link));
  });

  section.appendChild(h2);
  section.appendChild(ul);

  return section;
}
