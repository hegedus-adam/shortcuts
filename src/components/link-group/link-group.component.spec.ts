import { renderGroup } from './link-group.component';
import { JSDOM } from 'jsdom';

const mockRenderLink = jest.fn();
jest.mock('../link', () => ({
  renderLink: (link: any) => mockRenderLink(link),
}));

const dom = new JSDOM();
global.document = dom.window.document;
global.window = dom.window;

describe('renderGroup', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should append link', () => {
    const link = {
      id: 0,
      label: 'Test Label',
      url: 'https://somewhere.to',
    };

    const group = {
      groupLabel: 'Group Name',
      groupId: 0,
      links: [
        link,
      ],
    };

    const mockLink = document.createElement('li');
    mockLink.innerHTML = 'mock-link'
    mockRenderLink.mockReturnValue(mockLink);

    const actual = renderGroup(group);

    expect(actual).toMatchSnapshot();
    expect(mockRenderLink.mock.calls.length).toEqual(1);
    expect(mockRenderLink.mock.calls[0]).toEqual([link]);
  });
});
