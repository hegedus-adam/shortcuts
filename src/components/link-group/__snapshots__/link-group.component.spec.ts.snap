// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderGroup should append link 1`] = `
<section>
  <h2>
    Group Name
  </h2>
  <ul>
    <li>
      mock-link
    </li>
  </ul>
</section>
`;
